e2e/tests/cookie-banner.spec.ts

```diff
@@ -58,27 +80,81 @@ test.describe('cookie banner', () => {
-        test.skip(browserName === 'webkit', 'WebKit does not let you set Secure cookies on localhost')
+    test.describe('design-system-cookie-preference is set to 1', () => {
+      test.beforeEach(async ({ context, page }) => {
         await context.addCookies([{
           name: 'design-system-cookie-preference',
-          value,
+          value: '1',
           expires: Date.now() / 1000 + 1000 * 60 * 60,
           domain: 'localhost',
           path: '/',
         }])
+        await page.goto('/?cookie_prompt_enabled')
+      })
+
+      test('banner is hidden', async ({ page }) => {
+        await expect(page.getByRole('region', { name: 'Cookies on the MOD.UK Design System' })).toBeHidden()
+      })
+
+      test('analytics cookies are installed', async ({ browserName, page }) => {
+        test.skip(browserName === 'webkit', 'WebKit does not let you set Secure cookies on localhost')
+        expect(await page.evaluate(() => document.cookie)).toMatch(gaCookieRegex)
+      })
+    })
 
+    test.describe('design-system-cookie-preference is set to 0', () => {
+      test.beforeEach(async ({ context, page }) => {
+        await context.addCookies([{
+          name: 'design-system-cookie-preference',
+          value: '0',
+          expires: Date.now() / 1000 + 1000 * 60 * 60,
+          domain: 'localhost',
+          path: '/',
+        }])
         await page.goto('/?cookie_prompt_enabled')
+      })
+
+      test('banner is hidden', async ({ page }) => {
         await expect(page.getByRole('region', { name: 'Cookies on the MOD.UK Design System' })).toBeHidden()
+      })
 
-        const cookie = await page.evaluate(() => document.cookie)
-        expect(cookie).toMatch(`design-system-cookie-preference=${value}`)
+      test('analytics cookies are not installed', async ({ browserName, page }) => {
+        test.skip(browserName === 'webkit', 'WebKit does not let you set Secure cookies on localhost')
+        expect(await page.evaluate(() => document.cookie)).not.toContain('_ga')
       })
     })
   })
 
+  test.describe('cookie preferences expires', () => {
+    test('GA cookies are deleted if they are installed', async ({ context, page, browserName }) => {
+      test.skip(browserName === 'webkit', 'WebKit does not let you set Secure cookies on localhost')
+      await context.addCookies([{
+        name: '_ga',
+        value: '1',
+        expires: Date.now() / 1000 + 1000 * 60 * 60,
+        domain: 'localhost',
+        path: '/',
+      }, {
+        name: '_ga_TEST-ID',
+        value: '1',
+        expires: Date.now() / 1000 + 1000 * 60 * 60,
+        domain: 'localhost',
+        path: '/',
+      }])
+
+      await page.goto('/?cookie_prompt_enabled')
+      expect(
+        await page.evaluate(() => {
+          window.googleTagId = 'G-TEST-ID'
+          return document.cookie
+        }),
+      ).toEqual('_ga=1; _ga_TEST-ID=1')
```

Comment:
Seems like GOV.UK sites delete the GA cookies automatically in this case – what should we be doing?